diff --git a/CMakeLists.txt b/CMakeLists.txt
index e53102b..d858aae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,9 +17,15 @@ set(KF_MIN_VERSION "6.12.0")
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+option(SAILFISHOS "Build for Sailfish OS" OFF)
+
 find_package(ECM ${KF_MIN_VERSION} REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
 
+if (SAILFISHOS)
+    add_compile_definitions(Q_OS_SAILFISH)
+endif()
+
 include(KDECompilerSettings NO_POLICY_SCOPE)
 include(ECMAddTests)
 include(ECMGenerateHeaders)
@@ -51,8 +57,12 @@ if (ANDROID)
     find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS Kirigami)
     find_package(OpenSSL REQUIRED)
 else()
-    find_package(Qt6 ${QT_MIN_VERSION} REQUIRED COMPONENTS Widgets)
-    find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS QQC2DesktopStyle Crash)
+    if (SAILFISHOS) # Mix of Android and desktop
+        find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS Kirigami)
+    else()
+        find_package(Qt6 ${QT_MIN_VERSION} REQUIRED COMPONENTS Widgets)
+        find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS QQC2DesktopStyle Crash)
+    endif()
 endif()
 
 find_package(KPublicTransport 25.07.71)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index de75ec0..41594eb 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -63,7 +63,11 @@ if (ANDROID)
 
     ecm_add_android_apk(ktrip ANDROID_DIR ${CMAKE_SOURCE_DIR}/android)
 else()
-    target_link_libraries(ktrip PRIVATE Qt::Widgets KF6::Crash)
+    if (SAILFISHOS)
+        target_link_libraries(ktrip PRIVATE KF6::Kirigami)
+    else()
+        target_link_libraries(ktrip PRIVATE Qt::Widgets KF6::Crash)
+    endif()
 endif()
 
 ecm_add_app_icon(ktrip ICONS 256-org.kde.ktrip.png 150-org.kde.ktrip.png)
diff --git a/src/main.cpp b/src/main.cpp
index 4b9651b..93bcb97 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -15,10 +15,9 @@
 #include <QQuickStyle>
 #include <QTimer>
 
-#ifdef Q_OS_ANDROID
+#if defined Q_OS_ANDROID || defined Q_OS_SAILFISH
 #include <QGuiApplication>
 #else
-#include <KCrash>
 #include <QApplication>
 #endif
 
@@ -44,12 +43,19 @@ int main(int argc, char *argv[])
     QQuickStyle::setStyle(QStringLiteral("org.kde.breeze"));
 #else
     QIcon::setFallbackThemeName(QStringLiteral("breeze"));
+#ifdef Q_OS_SAILFISH
+    QGuiApplication app(argc, argv);
+    if (qEnvironmentVariableIsEmpty("QT_QUICK_CONTROLS_STYLE")) {
+        QQuickStyle::setStyle(QStringLiteral("org.kde.breeze"));
+    }
+#else
     QApplication app(argc, argv);
     // Default to org.kde.desktop style unless the user forces another style
     if (qEnvironmentVariableIsEmpty("QT_QUICK_CONTROLS_STYLE")) {
         QQuickStyle::setStyle(QStringLiteral("org.kde.desktop"));
     }
 #endif
+#endif
 
 #ifdef Q_OS_WINDOWS
     QApplication::setStyle(QStringLiteral("breeze"));
@@ -83,7 +89,7 @@ int main(int argc, char *argv[])
     about.setBugAddress("https://invent.kde.org/utilities/ktrip/-/issues");
     KAboutData::setApplicationData(about);
 
-#ifndef Q_OS_ANDROID
+#ifndef Q_OS_ANDROID && !defined(Q_OS_SAILFISH)
     QApplication::setWindowIcon(QIcon::fromTheme(QStringLiteral("org.kde.ktrip")));
     KCrash::initialize();
 #endif
diff --git a/src/org.kde.ktrip.desktop b/src/org.kde.ktrip.desktop
index d4948c8..ea2409a 100644
--- a/src/org.kde.ktrip.desktop
+++ b/src/org.kde.ktrip.desktop
@@ -75,7 +75,7 @@ GenericName[sv]=Kollektivtrafiknavigering
 GenericName[tr]=Toplu Taşıma için Yol Bulucu
 GenericName[uk]=Навігатор громадським транспортом
 GenericName[zh_TW]=公共運輸導覽工具
-Exec=ktrip
+Exec=qt6-start.sh ktrip
 Icon=org.kde.ktrip
 Type=Application
 Comment=Public transport navigator.
@@ -117,3 +117,8 @@ Comment[zh_TW]=公共運輸導覽工具。
 Terminal=false
 Categories=Qt;KDE;Network;
 X-KDE-FormFactor=desktop;tablet;handset;
+X-Nemo-Single-Instance=no
+X-Nemo-Application-Type=no-invoker
+
+[X-Sailjail]
+Sandboxing=Disabled
