diff --git a/src/org.kde.ktrip.desktop b/src/org.kde.ktrip.desktop
index b35c740..5c972f7 100644
--- a/src/org.kde.ktrip.desktop
+++ b/src/org.kde.ktrip.desktop
@@ -56,7 +56,7 @@ GenericName[sv]=Kollektivtrafiknavigering
 GenericName[tr]=Toplu Taşıma için Yol Bulucu
 GenericName[uk]=Навігатор громадським транспортом
 GenericName[x-test]=xxPublic transport navigatorxx
-Exec=ktrip
+Exec=qt-runner /usr/bin/ktrip
 Icon=org.kde.ktrip
 Type=Application
 Comment=Public transport navigator.
